app/features/modelling/baseReglModel.jsx

```diff
@@ -0,0 +1,85 @@
+import { Markdown } from 'markdownz';
+import React from 'react';
+import reglBase from 'regl';
+import alert from '../../lib/alert';
+
+class BaseReglModel {
+  constructor(canvas, { frame, metadata, sizing }) {
+    this.update = this.update.bind(this); // function to call on new annotation
+    this.getModel = this.getModel.bind(this);
+    this.calculateModel = this.calculateModel.bind(this);
+    if (this.getScore) {
+      this.getScore = this.getScore.bind(this); // (optional)
+    }
+
+    this.canvas = canvas;
+    this.regl = reglBase({
+      canvas,
+      attributes: { preserveDrawingBuffer: true },
+      extensions: ['OES_texture_float']
+    });
+
+    this.state = {
+      comparisonTexture: null,
+      dataHasLoaded: false,
+      metadata,
+      sizing,
+      frame,
+      pixels: this.regl.texture()
+    };
+    this.setModel();
+  }
+  update(annotations, newViewBox) {
+    // this function parses the new annotation and then triggers a render.
+    // update the render funtions
+    this.regl.poll();
+    this.regl.clear({
+      color: [0, 0, 0, 1]
+    });
+    this.state.pixels({
```

Comment:
If I'm reading this correctly, this appears to be a component state property that is a function? Does the returned result of the function become the state value? This is confusing to me and I'm wondering if this is directly mutating state, which is an anti-pattern in React.

Reply:
pixels is a reference to a WebGL texture
As this isn't a React component I've only used a variable called "state" to group things which are related to what the model should display (I think).